src/libstd/sys/unix/backtrace/mod.rs

```diff
@@ -91,15 +91,42 @@ mod tracing;
 // symbol resolvers:
 mod printing;
 
-#[cfg(not(any(target_os = "macos", target_os = "ios", target_os = "emscripten")))]
+#[cfg(not(target_os = "emscripten"))]
 pub mod gnu {
     use io;
     use fs;
     use libc::c_char;
 
+    #[cfg(not(any(target_os = "macos", target_os = "ios")))]
     pub fn get_executable_filename() -> io::Result<(Vec<c_char>, fs::File)> {
         Err(io::Error::new(io::ErrorKind::Other, "Not implemented"))
     }
+
+    #[cfg(any(target_os = "macos", target_os = "ios"))]
+    pub fn get_executable_filename() -> io::Result<(Vec<c_char>, fs::File)> {
```

Comment:
@alexcrichton the PR description talks about that:

> I've avoided calling that function to prevent having to play with the type of the returned filename, since the existing version calls for a null-terminated Vec<c_char> instead of a Vec<u8> with no null termination.